### PR TITLE
Add SecurityDefinition

### PR DESCRIPTION
**Pull request by msyyc**
modelerfour has supported security definition long time ago but testserver lacks swagger to test the feature.
autorest.python is developing the [feature](https://github.com/Azure/autorest.python/pull/1161) so it is more convenient to add the swagger which is also helpful to other language codegen tool

To test new swagger, please reference:
(1) security-key.json:
https://github.com/Azure/autorest.python/blob/ac12e56138ba92158aa52c547f829c0e339645b4/test/vanilla/legacy/AcceptanceTests/test_security.py#L40-L44

(2) seruity-aad.json
https://github.com/Azure/autorest.python/blob/ac12e56138ba92158aa52c547f829c0e339645b4/test/vanilla/legacy/AcceptanceTests/test_security.py#L35-L38

https://github.com/Azure/autorest.python/blob/ac12e56138ba92158aa52c547f829c0e339645b4/test/vanilla/legacy/AcceptanceTests/conftest.py#L89-L99

### CODE REVIEW

File: swagger/security_aad.json
Comment (iscai-msft) [inline swagger/security_aad.json:25]: super small nit: can we just make both operation ids `head`, I feel like there's no need to do an operation group

**Review by iscai-msft — APPROVED** [review]
looks great! Only thing is it's missing the actual server tests.

@timotheeguerin are we good to release this PR as is, and then add security tests later, or would you rather we do it in one go?

File: swagger/security_aad.json
Comment (timotheeguerin) [inline swagger/security_aad.json:1]: nit: name the files `kebab-case` to be consistent

**Comment by iscai-msft** [comment]
@msyyc I think once you get the CI passing, we can merge this PR with just the swagger fixes. During our scrum meeting I'll do a quick demo of how to write actual server tests (that include coverage). Then can you add actual server tests for these two examples? Thanks!

**Comment by timotheeguerin** [comment]
> @msyyc I think once you get the CI passing, we can merge this PR with just the swagger fixes. During our scrum meeting I'll do a quick demo of how to write actual server tests (that include coverage). Then can you add actual server tests for these two examples? Thanks!

Well technically here the CI is failing because new specs were added but not the mock api for it.

**Comment by iscai-msft** [comment]
oh @timotheeguerrin good point 😆 , @msyyc can you try to add tests as best as you can (and make sure they pass with your python tests)? We can go over it more during our weekly scrum

File: swagger/security_aad.json
Comment (msyyc) [reply swagger/security_aad.json:1]: got it

File: swagger/security_aad.json
Comment (msyyc) [reply swagger/security_aad.json:25]: ok

**Comment by msyyc** [comment]
> oh @timotheeguerrin good point 😆 , @msyyc Yuchao Yan FTE can you try to add tests as best as you can (and make sure they pass with your python tests)? We can go over it more during our weekly scrum

Have added test and it passes in my local environment: https://github.com/Azure/autorest.python/pull/1161#discussion_r824430429.

File: package-lock.json
Comment (iscai-msft) [inline package-lock.json:3]: @msyyc can you remove your commit for `package-lock.json`?

File: src/test-routes/security.ts
Comment (iscai-msft) [inline src/test-routes/security.ts:13]: I think it would be better to include the AAD test under the azure category

**Review by iscai-msft — APPROVED** [review]
yay overall looks really really good!

Agree about your point for AAD, it can be a little tricky. If you have time, I think you can try the following for your test:
1. Pass in a credential where you have overriden `get_token` to return something that won't change, maybe you can have it just return the jwt token made out of the scope that you pass to it. Here's a [resource](https://jwt.io/) to create your own jwt token (thanks @timotheeguerrin)
2. I believe the call for AAD should then have a header["Authorization"] that is a "Bearer ..." something, so you should be able to check for that on the server side too

I think we should hold off on merging this PR until the security PR is 100% ready, I think we're almost there. If we merge this PR without the security PR ready, we will run into "you don't have this test coverage" errors.

File: package-lock.json
Comment (msyyc) [reply package-lock.json:3]: reverted.

File: src/test-routes/security.ts
Comment (msyyc) [reply src/test-routes/security.ts:13]: For `vanilla`, I think it is also possible for AAD(like "webpubsub")?

**Review by iscai-msft — APPROVED** [review]
ok I think this PR is ready to go! going to wait until the security PR is in, I have one more comment in that PR, about why the constants tests were changed. Otherwise it looks good!

File: src/test-routes/security.ts
Comment (msyyc) [reply src/test-routes/security.ts:15]: good point

**Comment by iscai-msft** [comment]
@msyyc in the future, could you open a new PR if you're changing something unrelated to the original pr? we want to make sure our PRs are limited to the problem we want to solve, so combining security + the constants change here is not want we want. Next time they should be two separate PRs because they should not depend on each other at all. Thanks!

**Comment by msyyc** [comment]
@iscai-msft  Got it. I will pay attention to that in the future.

**Pull request merged**
